jscomp/syntax/src/res_core.ml

```diff
@@ -2251,6 +2264,64 @@ and parseBinaryExpr ?(context = OrdinaryExpr) ?a p prec =
 (* ) *)
 
 and parseTemplateExpr ?(prefix = "js") p =
+  let partPrefix =
+    match prefix with
+    | "js" | "j" | "json" -> Some prefix
```

Comment:
yeah something must be broken, I did this to make tests pass, but I agree it's ugly and there's no reason why we couldn't use ``` json`` ``` tagged templates, or `js` and `j` now that they don't make sense.

Reply:
This would be the relevant part:
```ml
  let parse_processed = function
    | None -> Some External_arg_spec.DNone
    | Some "json" -> Some DNoQuotes
    | Some "*j" -> Some DStarJ
```
basically, the delimiter determines how the string is emitted: whether in quotes, etc.
And this is going to interfere.